grype/matcher/matchers.go

```diff
@@ -81,9 +61,26 @@ func (c *controller) trackMatcher() (*progress.Manual, *progress.Manual) {
 	return &packagesProcessed, &vulnerabilitiesDiscovered
 }
 
-func (c *controller) findMatches(provider vulnerability.Provider, release *linux.Release, packages ...pkg.Package) match.Matches {
+func newMatcherIndex(matchers []Matcher) map[syftPkg.Type][]Matcher {
+	matcherIndex := make(map[syftPkg.Type][]Matcher)
+	for _, m := range matchers {
+		for _, t := range m.PackageTypes() {
+			if _, ok := matcherIndex[t]; ok {
```

Comment:
should this be `!ok`, it seems like it's going to make a new matcher array only if it finds one? maybe just remove this altogether, i think the append will do the right thing and you'll get `nil` if it's not found instead of a panic or something

Reply:
This is taken directly from the old controller code:
https://github.com/anchore/grype/blob/e77a6c8d63fe020d8f82c334947e2592a0567905/grype/matcher/controller.go#L57-L69

I'll double-check it to see if if should refactor it